mdata/cache/ccache_metric.go

```diff
@@ -137,24 +266,29 @@ func (mc *CCacheMetric) generateKeys() {
 // assumes we already have at least a read lock
 func (mc *CCacheMetric) nextTs(ts uint32) uint32 {
 	chunk := mc.chunks[ts]
-	span := chunk.Itgen.Span
+	return mc.nextTsCore(chunk.Itgen, chunk.Ts, chunk.Prev, chunk.Next)
+}
+
+// nextTsCore returns the ts of the next chunk, given a chunks key properties
+// (to the extent we know them). It guesses if necessary.
+// assumes we already have at least a read lock
+func (mc *CCacheMetric) nextTsCore(itgen chunk.IterGen, ts, prev, next uint32) uint32 {
```

Comment:
afaict there is no case where `itgen.Ts != ts`, so those two arguments are redundant